### DISCUSSION TITLE
Browser Fingerprinting

### DISCUSSION THREAD

**Discussion by ghost** [minbrowser/min#2288] [Ideas]
Fingerprinting is a way of identifying your web browser (device, etc.), whether you're logged in or not.

Potential vulnerabilities:

+ Canvas API
+ HTTP Headers
+ Geolocation
+ WebRTC
+ Fonts

Fingerprinting tests:

+ [EFF](https://coveryourtracks.eff.org)
+ [Privacy.net](https://privacy.net/analyzer)
+ [Browserleaks](https://browserleaks.com)

**Comment by giuseppe-macario** [comment]
Browser fingerprinting makes extensive use of JavaScript. Therefore, by disabling JavaScript, you can block most fingerprinting techniques right away.

On the other hand, if you were requesting some form of active blocking, i.e. randomized or spoofed fingerprint, I doubt a minimalist browser such as Min could implement those algorithms, which are out of scope and require a huge amount of code/work. You should consider Brave: being developed by the inventor of JavaScript (Brendan Eich), it is now the best non-minimalist browser that combats fingerprinting, just because no one in the world knows JavaScript — and its risks — better than its inventor.

**Reply by xplshn** [reply]
Its not that complicated, just change the user agent for all users to be the same and add extensions support so users who want better privacy can use extensions like adnauseam, and others.

**Reply by giuseppe-macario** [reply]
In this case, changing the user agent is pretty useless, because Min browser already has a "generic" user agent (on purpose, I suppose).
As for extensions, they can't block fingerprinting, apart from those that block scripts… in which case you could also disable JavaScript yourself.
